cpp/src/io/parquet/page_enc.cu

```diff
@@ -1938,13 +1942,13 @@ dremel_data get_dremel_data(column_view h_col,
  *
  * @param[in,out] frag Fragment array [column_id][fragment_id]
  * @param[in] col_desc Column description array [column_id]
- * @param[in] num_fragments Number of fragments per column
+ * @param[in] num_fragments Number of fragments per column, -1 if fragment sizes already specified
  * @param[in] num_columns Number of columns
  * @param[in] stream CUDA stream to use, default 0
  */
 void InitPageFragments(device_2dspan<PageFragment> frag,
                        device_span<parquet_column_device_view const> col_desc,
-                       uint32_t fragment_size,
+                       int32_t fragment_size,
```

Comment:
Are we concerned about the loss of resolution here?

Reply:
I asked a similar question on another PR and @vuule said it's fine. https://github.com/rapidsai/cudf/pull/9677#discussion_r748668781